builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,289 @@
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsElasticacheReplicationGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	replicationGroupId := d.Get("replication_group_id").(string)
+	description := d.Get("description").(string)
+	cacheNodeType := d.Get("cache_node_type").(string)
+	automaticFailover := d.Get("automatic_failover").(bool)
+	numCacheClusters := d.Get("num_cache_clusters").(int)
+	engine := d.Get("engine").(string)
+	engineVersion := d.Get("engine_version").(string)
+	securityNameSet := d.Get("security_group_names").(*schema.Set)
+	securityIdSet := d.Get("security_group_ids").(*schema.Set)
+
+	securityNames := expandStringList(securityNameSet.List())
+	securityIds := expandStringList(securityIdSet.List())
+
+	req := &elasticache.CreateReplicationGroupInput{
+		ReplicationGroupID:		aws.String(replicationGroupId),
+		ReplicationGroupDescription:	aws.String(description),
+		CacheNodeType:            	aws.String(cacheNodeType),
+		AutomaticFailoverEnabled: 	aws.Boolean(automaticFailover),
+		NumCacheClusters:         	aws.Long(int64(numCacheClusters)),
+		Engine:				aws.String(engine),
+		EngineVersion:			aws.String(engineVersion),
+		CacheSecurityGroupNames:	securityNames,
+		SecurityGroupIDs:		securityIds,
+	}
+
+	if v, ok := d.GetOk("parameter_group_name"); ok {
+		req.CacheParameterGroupName = aws.String(v.(string))
+	}
+
+	_, err := conn.CreateReplicationGroup(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache replication group: %s", err)
+	}
+
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    ReplicationGroupStateRefreshFunc(conn, d.Id(), "available", pending),
+		Timeout:    15 * time.Minute,
+		Delay:      20 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be created: %s", d.Id(), sterr)
+	}
+
+	d.SetId(replicationGroupId)
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	req := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	res, err := conn.DescribeReplicationGroups(req)
+	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "ReplicationGroupNotFoundFault" {
+			// Update state to indicate the replication group no longer exists.
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	if len(res.ReplicationGroups) == 1 {
+		c := res.ReplicationGroups[0]
+		d.Set("replication_group_id", c.ReplicationGroupID)
+		d.Set("description", c.Description)
+		d.Set("automatic_failover", c.AutomaticFailover)
+		d.Set("num_cache_clusters", len(c.MemberClusters))
```

Comment:
Having access to the primary endpoint address is really useful.  As I will typically take the address and create a cname record in Route53, which my instances can point to.

A little hacky, but it'd look something like this.
```
if (len(c.NodeGroups) == 1) {
  d.Set("primary_endpoint", c.NodeGroups[0].PrimaryEndpoint.Address)
}
```